polkadot/node/core/backing/src/lib.rs

```diff
@@ -649,7 +644,7 @@ fn table_attested_to_backed(
 			.map(|(pos_in_votes, _pos_in_group)| validity_votes[pos_in_votes].clone())
 			.collect(),
 		validator_indices,
-		inject_core_index.then_some(core_index),
+		Some(core_index),
```

Comment:
This can no longer be None, so we need to refactor it as well, together with the functions on `BackedCandidate` which handle the injected core index, like `validator_indices_and_core_index` and `set_validator_indices_and_core_index`